src/Compilers/VisualBasic/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.vb

```diff
@@ -0,0 +1,300 @@
+            Return AddDynamicAnalysis(original, MyBase.CreateWithStatementPrologue(original))
+        End Function
+
+        Private Function AddDynamicAnalysis(original As BoundStatement, rewritten As BoundStatement) As BoundStatement
+            If Not original.WasCompilerGenerated Then
+                Return CollectDynamicAnalysis(original, rewritten)
+            End If
+
+            Return rewritten
+        End Function
+
+        Private Function CollectDynamicAnalysis(original As BoundStatement, rewritten As BoundStatement) As BoundStatement
+            Dim statementFactory As New SyntheticBoundNodeFactory(_factory.TopLevelMethod, _method, original.Syntax, _factory.CompilationState, _diagnostics)
+            Return statementFactory.Block(ImmutableArray.Create(AddAnalysisPoint(SyntaxForSpan(original), statementFactory), rewritten))
+        End Function
+
+        Private Function AddAnalysisPoint(syntaxForSpan As VisualBasicSyntaxNode, statementFactory As SyntheticBoundNodeFactory) As BoundStatement
+            ' Add an entry in the spans array.
+
+            Dim location As Location = syntaxForSpan.GetLocation()
+            Dim spanPosition As FileLinePositionSpan = location.GetMappedLineSpan()
+            Dim path As String = spanPosition.Path
+            If path.Length = 0 Then
+                path = syntaxForSpan.SyntaxTree.FilePath
+            End If
+
+            Dim spansIndex As Integer = _spansBuilder.Count
+            _spansBuilder.Add(New SourceSpan(_debugDocumentProvider.Invoke(path, ""), spanPosition.StartLinePosition.Line, spanPosition.StartLinePosition.Character, spanPosition.EndLinePosition.Line, spanPosition.EndLinePosition.Character))
+
+            ' Generate "_payload(pointIndex) = True".
+
+            Dim payloadCell As BoundArrayAccess = statementFactory.ArrayAccess(statementFactory.Local(_methodPayload, False), True, statementFactory.Literal(spansIndex))
+            Return statementFactory.Assignment(payloadCell, statementFactory.Literal(True))
+        End Function
+
+        Private Shared Function SyntaxForSpan(statement As BoundStatement) As VisualBasicSyntaxNode
+            SyntaxForSpan = statement.Syntax
+
+            Select Case statement.Kind
+                Case BoundKind.IfStatement
+                    SyntaxForSpan = DirectCast(statement, BoundIfStatement).Condition.Syntax
+                Case BoundKind.WhileStatement
+                    SyntaxForSpan = DirectCast(statement, BoundWhileStatement).Condition.Syntax
+                Case BoundKind.ForEachStatement
+                    SyntaxForSpan = DirectCast(statement, BoundForEachStatement).Collection.Syntax
+                Case BoundKind.DoLoopStatement
+                    Dim condition As BoundExpression = DirectCast(statement, BoundDoLoopStatement).ConditionOpt
+                    If condition IsNot Nothing Then
+                        SyntaxForSpan = condition.Syntax
```

Comment:
Yes, because the instrumentation is actually of the condition. If there is no condition, the statement shouldn't be instrumented. I'll make that change now, and be sure to test it as part of the process of adding real testing.